examples/nidcpower_source_dc_voltage/teststand_fixture.py

```diff
@@ -50,32 +52,12 @@ def create_nidcpower_sessions(sequence_context: Any) -> None:
         pin_map_id = teststand_support.get_active_pin_map_id()
 
         pin_map_context = nims.session_management.PinMapContext(pin_map_id=pin_map_id, sites=None)
-        with session_management_client.reserve_sessions(
-            context=pin_map_context,
-            instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
-            # This code module sets up the sessions, so error immediately if they are in use.
-            timeout=0,
-        ) as reservation:
+        with reserve_session(session_management_client, pin_map_context, timeout=0) as reservation:
             for session_info in reservation.session_info:
-                options: Dict[str, Any] = {}
-                if USE_SIMULATION:
-                    options["simulate"] = True
-                    options["driver_setup"] = {"Model": "4141"}
-
-                grpc_options = nidcpower.GrpcSessionOptions(
-                    grpc_channel_pool.get_grpc_device_channel(
-                        nidcpower.GRPC_SERVICE_INTERFACE_NAME
-                    ),
-                    session_name=session_info.session_name,
-                    initialization_behavior=nidcpower.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
-                )
-
-                # Leave session open
-                nidcpower.Session(
-                    resource_name=session_info.resource_name,
-                    options=options,
-                    grpc_options=grpc_options,
+                grpc_channel = grpc_channel_pool.get_grpc_device_channel(
+                    nidcpower.GRPC_SERVICE_INTERFACE_NAME
                 )
```

Comment:
Hmm, you can't reuse `_helpers.get_grpc_device_channel` because it expects a `MeasurementService`.

What if it took a `Union[MeasurementService, ChannelPool]`? Then you could use it in both places.

Reply:
you are right Brad. We can't reuse `_helpers.get_grpc_device_channel` in teststand code.

The reason for that, teststand code uses the `get_grpc_device_channel` from ChannelPool, and that inturn calls the method below.

 ```
def get_grpc_device_channel(self, provided_interface: str) -> grpc.Channel:
        """Return gRPC channel to specified NI gRPC Device service.

        Args:
            provided_interface (str): The gRPC Full Name of the service.

        """
        return self.get_channel(
            self._discovery_client.resolve_service(
                provided_interface=provided_interface,
                service_class="ni.measurementlink.v1.grpcdeviceserver",
            ).insecure_address
        )


```

Reply:
The TestStand code uses ChannelPoolHelper, not ChannelPool, so let's update that to expect a driver module.

```diff
diff --git a/examples/nidcpower_source_dc_voltage/_helpers.py b/examples/nidcpower_source_dc_voltage/_helpers.py
index d0bdc94..1ef4fe7 100644
--- a/examples/nidcpower_source_dc_voltage/_helpers.py
+++ b/examples/nidcpower_source_dc_voltage/_helpers.py
@@ -109,16 +109,16 @@ class GrpcChannelPoolHelper(GrpcChannelPool):
             ).insecure_address
         )

-    def get_grpc_device_channel(self, provided_interface: str) -> grpc.Channel:
+    def get_grpc_device_channel(self, driver_module: types.ModuleType) -> grpc.Channel:
         """Return gRPC channel to specified NI gRPC Device service.

         Args:
-            provided_interface (str): The gRPC Full Name of the service.
+            driver_module: The driver module (e.g. nidcpower)

         """
         return self.get_channel(
             self._discovery_client.resolve_service(
-                provided_interface=provided_interface,
+                provided_interface=getattr(driver_module, "GRPC_SERVICE_INTERFACE_NAME"),
                 service_class="ni.measurementlink.v1.grpcdeviceserver",
             ).insecure_address
         )
diff --git a/examples/nidcpower_source_dc_voltage/teststand_fixture.py b/examples/nidcpower_source_dc_voltage/teststand_fixture.py
index 73aa3ee..aa0ad9b 100644
--- a/examples/nidcpower_source_dc_voltage/teststand_fixture.py
+++ b/examples/nidcpower_source_dc_voltage/teststand_fixture.py
@@ -83,9 +83,7 @@ def destroy_nidcpower_sessions() -> None:

             for session_info in reservation.session_info:
                 grpc_options = nidcpower.GrpcSessionOptions(
-                    grpc_channel_pool.get_grpc_device_channel(
-                        nidcpower.GRPC_SERVICE_INTERFACE_NAME
-                    ),
+                    grpc_channel_pool.get_grpc_device_channel(nidcpower),
                     session_name=session_info.session_name,
                     initialization_behavior=nidcpower.SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION,
                 )
```

Reply:
Oops, teststand_fixture.py has two calls that create the session. We should update both.